Make mpFaultTolerance 4.0 tolerate MP 6 and EE 10

Azquelt opened the pull request:
mpFaultTolerance-4.0 will be included in both MP 5 and MP 6

For #21968

tevans78 committed:
Make mpFaultTolerance 4.0 tolerate MP 6 and EE 10

mpFaultTolerance-4.0 will be included in both MP 5 and MP 6

It needs to work with both mpMetrics-4.0 and mpMetrics-5.0

tevans78 committed:
Create mpMetrics-5.0 stub feature

Azquelt committed:
Repeat MP FT tests for MP 6.0

Azquelt commented:
Not passing yet as MP Fault Tolerance requires metrics. We could merge this without tests, or with metrics tests disabled for MP 6.0 depending on how long it is going to be before mpMetrics-5.0 is usable.

Metrics isn't going to be ready soon so I've added a stub mpMetrics-5.0 feature which uses the mpMetrics-4.0 implementation but depends on EE10 and MP6.

Azquelt committed:
Add mpMetrics to the MP60 feature set

Azquelt committed:
Update FT+RestClient integration FAT for MP60

Update the MP Fault Tolerance and MP Rest Client integration tests to
also run against MicroProfile 6 + EE10

Azquelt commented:
#libby #build

Azquelt committed:
Fix EE9 Action to remove EE10 features

Azquelt commented:
#build
#libby

tevans78 commented on dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat/fat/src/com/ibm/websphere/microprofile/faulttolerance_fat/multimodule/tests/TestMultiModuleClassLoading.java:
```diff
@@ -46,10 +46,12 @@ public class TestMultiModuleClassLoading extends FATServletClient {
     @ClassRule
-    public static RepeatTests r = RepeatFaultTolerance.repeat(SERVER_NAME, TestMode.LITE, MicroProfileActions.MP50, MicroProfileActions.MP13,
-                                                              RepeatFaultTolerance.MP21_METRICS20);
+    public static RepeatTests r = RepeatFaultTolerance.repeat(SERVER_NAME, TestMode.LITE, MicroProfileActions.MP60,
```
are you sure that all 4 repeats are needed in LITE?

Azquelt replied:
Spent a while looking at why this might be before I realised that the whole test has `@Mode(FULL)`.

I'll remove `LITE` here (and elsewhere) since it has no effect and is confusing.

Azquelt replied:
Done

Azquelt commented:
#libby #build

Azquelt commented:
Build failed due to a bug introduced in #22131 which has since been reverted. Unfortunately it prevented some of the fault tolerance tests from running so I'll need to rebase on integration and run it again.

Azquelt commented:
#libby #build

posmith commented:
L2 message review not required

Azquelt commented:
Rebase to remove merge conflict
#build
#libby

Azquelt commented:
#build
#libby

Azquelt committed:
Make metrics-5.0 API feature depend on noShip

Avoids VisibilityTest complaining that a non-beta feature depends only
on beta features.

Azquelt commented:
Rebase again to avoid another merge conflict.

Rather than changing `VisibilityTest`, I've made the metrics API feature depend on the `noShip` feature.

#libby
#build

Azquelt commented:
Failures in the build do not appear to be related to this change.
